common/src/main/java/org/conscrypt/ConscryptEngine.java

```diff
@@ -1081,39 +1215,64 @@ private SSLEngineResult readPendingBytesFromBIO(ByteBuffer dst, int bytesConsume
      */
     private int readEncryptedData(final ByteBuffer dst, final int pending) throws SSLException {
         try {
-            int bioRead = 0;
+            int bytesRead = 0;
+            final int pos = dst.position();
```

Comment:
@flooey we don't have to worry about multithreading at this point, since we're already synchronized here: https://github.com/google/conscrypt/blob/master/common/src/main/java/org/conscrypt/ConscryptEngine.java#L676